src/components/CustomRadioButton/CustomRadioButton.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { bool, string } from 'prop-types';
+
+import { radioOption } from '../../types';
+import InputRadio from './InputRadio';
+import LabelIcon from './LabelIcon';
+import LabelText from './LabelText';
+
+const renderLabel = (label, id, withIcon) =>
+  withIcon ? <LabelIcon htmlFor={id}>{label}</LabelIcon> : <LabelText htmlFor={id}>{label}</LabelText>;
+
+const CustomRadioButton = ({ radio, name, withIcon }) => {
+  const { value, label, id } = radio;
+  return (
+    <>
+      <InputRadio type="radio" name={name} value={value} id={id} />
+      {renderLabel(label, id, withIcon)}
+    </>
+  );
+};
+
+CustomRadioButton.propTypes = {
+  radio: radioOption.isRequired,
+  name: string.isRequired,
+  withIcon: bool.isRequired
```

Comment:
It is not a good approach to make all props required. Instead thinks if you want to provide that property everytime you need to render that component? What is the default component shape? Maybe `withoutIcon` can be set as default? If yes, just provide default value for that instead of requirement during every render.